src/Riok.Mapperly/Descriptors/Mappings/LinqDictionaryMapping.cs

```diff
@@ -0,0 +1,55 @@
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+using static Riok.Mapperly.Emit.SyntaxFactoryHelper;
+
+namespace Riok.Mapperly.Descriptors.Mappings;
+
+/// <summary>
+/// Represents an enumerable mapping which works by using linq (select + collect).
+/// </summary>
+public class LinqDicitonaryMapping : TypeMapping
+{
+    private const string LambdaParamName = "x";
+
+    private const string KeyValueKeyPropertyName = nameof(KeyValuePair<object, object>.Key);
+    private const string KeyValueValuePropertyName = nameof(KeyValuePair<object, object>.Value);
+
+    private readonly IMethodSymbol _collectMethod;
+    private readonly ITypeMapping _keyMapping;
+    private readonly ITypeMapping _valueMapping;
+
+    public LinqDicitonaryMapping(
+        ITypeSymbol sourceType,
+        ITypeSymbol targetType,
+        IMethodSymbol collectMethod,
+        ITypeMapping keyMapping,
+        ITypeMapping valueMapping)
+        : base(sourceType, targetType)
+    {
+        _collectMethod = collectMethod;
+        _keyMapping = keyMapping;
+        _valueMapping = valueMapping;
+    }
+
+    public override ExpressionSyntax Build(TypeMappingBuildContext ctx)
+    {
+        var lambdaParamName = ctx.NameBuilder.New(LambdaParamName);
+
+        // if key and value types do not change then use a simple call
+        if (SymbolEqualityComparer.Default.Equals(_keyMapping.TargetType, _keyMapping.SourceType)
+            && SymbolEqualityComparer.Default.Equals(_valueMapping.TargetType, _valueMapping.SourceType))
```

Comment:
Instead of checking the type, it should be ok to check `IsSynthetic` on the mapping.

Reply:
I meant to ask about this,, for some reason `_keyMapping` and `_valueMapping` `IsSynthetic` kept returning true.

Reply:
`IsSynthetic` is truethy if the mapping does not produce any code and could be omitted. Do you observed `IsSynthetic` beeing true for a mapping which produces code?